fix a crash in base64 buffer pool

harshavardhana opened the pull request:
## Description
fix a crash in base64 buffer pool

## Motivation and Context
looks like 1024 buffer size is not enough in
all situations, use 8192 instead which
can satisfy all the rare situations that
may arise in base64 decoding.

## How to test this PR?
Reproduced using warp GET benchmark in a distributed setup

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression probably yes but this is a functionality change anyways introduced in bfe8a9bccccb86233bf627f4627c724a70e4a188
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

klauspost reviewed (approved):
I don't really understand the issue. The decoder decodes into the buffer, but the buffer must be DecodedLen(len(input)) not to fail. I presume that is what is failing when the JSON is too large?

I don't mind the change. Just want to understand the background.

harshavardhana committed:
fix a crash in base64 buffer pool

looks like 1024 buffer size is not enough in
all situations, use 8192 instead which
can satisfy all the rare situations that
may arise in base64 decoding.
